library/Garp/Db/Table/Rowset.php

```diff
@@ -82,4 +82,19 @@ public function filter($fn): Garp_Db_Table_Rowset {
         return $out;
     }
 
+    /**
+     * Reduce a rowset by $fn to $initial
+     *
+     * @param callable $fn
+     * @param mixed $initial
+     * @return mixed
+     */
+    public function reduce(callable $fn, $initial) {
+        $result = $initial;
+        foreach ($this->toArray() as $row) {
```

Comment:
Why have you chosen to use `toArray()` here? The callback could also be passed the row object.
(not consistent with the other functional methods, I know, but I should never have done that)

Reply:
You're right it must by `toArray()`. I didn't think of it. The only ugly thing is that you have to call `toArray()` on the row when passing it on to the rowset.